### ISSUE TITLE
M99.0.4803.0 randomly crashes

### ISSUE THREAD

**Issue by TrainzLuvr** [Alex313031/Thorium-Win#3]
Hi,

Running on Win 10, using Chromium Launcher, browser profile is on a network drive. Launcher command line is:

`ChromiumCommandLine=--user-data-dir="W:\xxxxx" --no-default-browser-check --allow-outdated-plugins --disable-logging --disable-breakpad --enable-tab-audio-muting --show-component-extension-options --disable-ipv6 --disk-cache-dir="C:\xxxxx"`

If I walk away from the desktop and come back hour or so later, I'd notice the browser is gone. When I reload it, says it crashed and it restores all my open tabs. This last time I had about 40 tabs but shouldn't matter as I have enough RAM in the system.

On at least one occasion it also crashed on startup. I remember clicking the icon and the browser loaded, then few moments later it was just gone. I thought to myself whether I actually clicked the icon or not, but I know I did, so it was a crash.

This was not happening with a regular Chromium or any of the other optimized flavours.

Thanks.

**Comment by Alex313031** [comment]
hmmmm. I havent used this release extensively outside of testing it in a win11 and win7 VM. It might be an upstream bug, or the fact the profile and cache? are on a network drive. I will be making a new release soon that you can test.

**Comment by TrainzLuvr** [comment]
Thanks.

I still keep the cache on the local drive, and I've never experiencied any issues with the profile being on the network drive (aside from Chromium sometimes complaining that's the case).

It might be an upstream bug too, or win10 (21H2 19044.1499) issue...

**Comment by Alex313031** [comment]
OH one thing it might be. Go to wherever the thorium executable is (thorium.exe in your portable folder or C:\Users\$USERNAME\AppData\Local\Chromium\chrome.exe), right click and go to properties, then compatability, and then set the compatibility mode to windows 8. Then relaunch.

**Comment by TrainzLuvr** [comment]
I think I'll wait on the next build because I've been experiencing some other weird behaviours: clicks not registering, downloads not starting, basically user actions not performing.

Unfortunately, it's been random so I cannot provide details on how to reproduce.

**Comment by Alex313031** [comment]
@TrainzLuvr OK. That thing about compatibility mode though, you will still want to set that. And want to set it on the next release. This has been a known bug upstream for 2 major versions now!!

**Comment by Alex313031** [comment]
@TrainzLuvr 100.0.4848.0 is out now, and the compatibility thing is fixed!

**Comment by TrainzLuvr** [comment]
Awesome, will give it a try.

Thanks!

**Comment by Alex313031** [comment]
@TrainzLuvr How did it work out for you?

**Comment by TrainzLuvr** [comment]
It didn't - after a few hours it crashed. :(

I don't know what's going on anymore, maybe it's some extension causing the crash because I have not had crashes in the past, though I used an older version of Chromium (v81). Only since I updated to v9x did this start happening.

**Comment by Alex313031** [comment]
@TrainzLuvr So this isnt just with Thorium? I use almost all of my releases as my daily driver browser, and leave it open for sometimes days at a time with literally hundreds of tabs open. It may be an extension. Tell me what extensions you have, and what your OS is and maybe I can help.
Also, as a last resort, it may be a user profile corruption thing. In that case, I would go to chrome://extensions and take a picture of all the extensions you have. Then go to chrome://bookmarks and export your bookmarks to an html file, then install the session buddy extension to save and export your session if you have lotsa tabs open, then you can find extensions to back up your history as a xml file. Then go to ~/.config/thorium or ~/.config/chromium or [where you extracted thorium portable]/.profile and delete everything in there and start from scratch, using the above mentioned to restore all pertinent data.

**Comment by TrainzLuvr** [comment]
I've used an older Chromium (v81) for a long time and never had any problems at all. I did not change any of my extensions during the upgrade to v9x, and so unless some extension was also updated to cause the problem, to me everything points back to v9x Chromium series as a culprit.

I have my profile on the network drive, while cache is local, but that has never been an issue with previous Chromiums (< v9x) and it really surprised me to see the window just disappear, on startup even, as well as hours into a daily session. Each time it disappeared I did not have many tabs open, matter a fact maybe 3-4 tops. I'd be really upset if I had my usual 50+ tabs open and it tanked on me. :(

I'm using Windows 10 (21H2), with extensions: uBlock Origin, uBlock Origin Extra, Privacy Badger, Tampermonkey, Edit This Cookie, User JavaScript and CSS, Google Images Restored, Old Reddit Redirect, Skip Redirect, BlockTube, uBlacklist, Clickbait Remover for YouTube, Lan port scan forbidder...

I wish they made some kind of a tool to check/verify the browser profile for corruption but I suppose browsers are so complex  that doing this might not be possible, or the Chromium devs might just say to start with a new profile instead. I'll give a try what you suggested with exporting pertinent data and making a new profile when I get a chance.

TBH, it's 2022 and it's been decades since we had anything remotely close to a true network browser profile. The last one that was truly good was in Netscape Communicator 4.8 which was awesome and could follow you around anywhere you went. Oh well, I can only lament about these things now (aside from dating myself, LOL).

**Comment by Alex313031** [comment]
Chromium 81 is quite old, and anything older than M86 is vulnerable to some attacks that are actively being exploited in the wild.
I use alot of those extensions, and im gonna try blocktube out. ublock origin extra is sadly discontinued and now serves no purpose, and I would reccomend uninstalling it. Read the top of the readme in the now archived github > https://github.com/gorhill/uBO-Extra#readme

As for using a network drive for profile, this may be causing it, so may I ask why you put it on a network drive? The user profile dir has never gotten above 2GB for me, and I have 20 extensions and usually hundreds of tabs open.

You may want to set the user policy RoamingProfileSupportEnabled and RoamingProfileLocation to your network drive rather than just --user-data-dir, which may be causing your issues. See here > https://support.google.com/chrome/a/answer/7349337 and here > https://www.reddit.com/r/sysadmin/comments/a9xbpz/google_chrome_user_profiles_network_locations_and/

Also what do you mean by "true network browser profile" & "could follow you around anywhere you went"?

In regards to netscape, I'm 21 and grew up with XP, yet Im quite the outlier for my age, as Im a legacy os/tech enthusiast. I have ALOT of computers at my house in general, but I have a IBM Ps/2 with dos 6.22 and win 3.11, a pentium 3 with win 95, a celeron with win nt4.0, and a pentium 4 with win 98se and win 2000, and a core 2 duo with xp. I consider core 2 duo to be the upper limit of "legacy" I got some cool boxed software including wordperfect, excel for mac II, xp home edition with geeksquad sticker from before they were bought by bestbuy, and win nt 4.0 oem with certificate and boot floppies, various software and games on cd and floppy. I have the web working on win 98se and win 2000, and kinda sorta for http only sites on the win95 machine. For my main machine I have a 4790k, 24gb ram, z97 board, and gtx 970 with 3 1080p dell monitors, and a genuine IBM ps/2 keyboard, and a (very hard to find because they are almost always ball mice since the time periods dont really align) ps/2 LASER mouse.
I run windows xp, windows 7 as main win drive, windows 10 (only for development and games that require it), win 11, and ubuntu, arch linux, and crunchbangplusplus (debian based and my daily driver/main dev drive), and chromiumos and cloudready. I make the chromiumos builds myself > https://github.com/Alex313031/ChromiumOS

OML i went way longer with that than i shoulda. 
I would reccomend, setting those things above for your network drive. If it still fails, I would rename the user profile, and start with a fresh one. If that resolves it, then restore the old profile, backup everything, and create new. If none of this works, I would copy your user profile temporarily to the local disk and point it there with --user-data-dir. If still none of this works, I would try a stable chromium release. If STILL none of this works, then I would say somethings strange with your machine/how your os is set up.

**Comment by Alex313031** [comment]
@TrainzLuvr  Oh, and is there a constant, uninterrupted connection to the network drive. If latency or jitter is high, or the machine with the network drive suspends itself, or any processes, that might also cause it. This is because chromium uses a lock file that it periodically rewrites, even if the browser is not doing anything at all and nothing else is being written/read to the user profile dir, this is to check for exclusive access and that the user profile isnt modified by another chromium process or malicious program. You can see this for yourself if you run a chromium window, then download a portable of another version of chromium and try to open it, it will just launch another window from the previous chromium process. To run two separate instances of chromium, you have to set seperate user data dirs. My linux builds use ~/.config/thorium rather than ~/.config/chromium, and of course the portable win versions have a bat that disables logging, registry entries, and sets the user profile to a dir in the extracted zip. Work is ongoing to set thorium in source code to use %APPDATA%\Local\Thorium instead of chromium to allow installs with the exe to put the program's files and user data dir to seperate dirs from chromium to allow someone to install them side by side easily.

**Comment by TrainzLuvr** [comment]
As I said, this was never happening with v81, despite all the extensions, despite the profile on the network (LAN), it was all rock solid and reliable. Now, I can't say anymore, but unfortuantely there's nowhere else to go either. I wish I could go back to v86 but unfortunately, that's not possible with these profiles, I found that once you upgrade they are tainted permanently it seems. 

I meant Netscape Communicator's ROAMING profile, once you set it up, and it saved your bookmarks, history, cookies, etc on the WebDAV you could go anywhere and connect to it. A side benefit was that you could access bookmarks and cookies directly from WebDAV via HTTP and read them without the browser. Good old days, long gone. :(

You have a nice collection of legacy hardware. Especially that we both like the mechanical IBM keyboards - nothing can replace that click (I'm using an Apple Aluminum KBD right now and it feels "squishy" to me compared to the IBM click). My collection goes further back though: I got a few SGIs, couple of Amigas (with VT), but true gems (to me) are Truevision ATVista and Targa graphics cards (EISA). :D

I'm going to try all of your suggestions above when I get a chance and report back. Everything has gotten overly complicated imho and we are spending way too much time managing the OS and its apps, and not creating things, as we should be.

**Comment by Alex313031** [comment]
@TrainzLuvr Cool about netscape. And yeah, nothing compares to a buckling spring switch. I've always wanted an Amiga. Those cards are OLD lolol in terms of GPUs. Yeah, lemme know how that all works for you with Thorium.

Are you a lover of trains? Because besides my obsession with computers/FOSS/Linux/Chromium, I love steam engines, both stationary and locomotives, but especially locomotives. Im also obsessed with the Titanic, and recently completed a model of it.

**Comment by TrainzLuvr** [comment]
Yeah I like trains. :D ... also building a model railroad layout in my basement. Couldn't do steam in N Scale because it doesn't have the weight to pull many cars, so had to settle for more modern diesels.
I've seen people build miniature working models of steam engines, which was nuts. Also like those other railorad modelers who build their own live steam locomotives and ride on them. That would be awesome, but I don't have the kind of resources or space (my backyard might only fit a G loop at best), LOL.

**Comment by Alex313031** [comment]
@TrainzLuvr Hows it coming?

**Comment by TrainzLuvr** [comment]
Thanks for following-up

No change, it still crashes, now within a minute or two of launch, running v100.0.4848.0.

Though I haven't had a chance to move to a local disk, I'm still mulling over it, meh.

**Comment by Alex313031** [comment]
@TrainzLuvr How is it coming. New releases are out, as well as new AVX2 repo.

**Comment by TrainzLuvr** [comment]
I went back to Chromium v98 because none of my bank sites liked v100+ and I couldn't log in.

Wish I could go back to v86/87/88/89 because they were the ones that worked the best without any crashes for me.

**Comment by Alex313031** [comment]
@TrainzLuvr I actually made a M89 release with all the compiler optimizations of Thorium for @raingart here > https://github.com/Alex313031/Snippets/releases/tag/chromium

**Comment by TrainzLuvr** [comment]
I don't think running M89 is a good idea now with CVE-2022-1096.

Also, all of my settings have been upgraded to M99 now, so it's almost impossible to go back. It would take an unreasonable amount of time to reverse the DB changes.

**Comment by Alex313031** [comment]
@TrainzLuvr What would you like me to do? Your issues would probably be fixed if you used a local profile. I don't understand your aversion to it, chromium hard limits it to 2 GB and I have literally 1835 tabs open in 34 windows and it still only is using 1.2GB. Your profile would be under a gig. 
I don't really know how to help from here, if you don't wanna do a local profile dir and don't wanna downgrade. You could stay on M98, but otherwise.

**Comment by TrainzLuvr** [comment]
Oh no, please don't feel you need to do something, I have no expectations here.

My Chromium folder which contains the profile is 1.82GB right now. It does not include any cache files, as I store that locally.

But this is my setup, and I run it this way because I do not trust Windows or any of my SSD drives, to ever be reliable 100%. So I put everything I can onto my LAN where I have redundant storage, and feel somewhat safe I won't lose day-to-day data.

We can close this issue if you wish, as Chromium clearly does not like network storage.

**Comment by Alex313031** [comment]
@TrainzLuvr That's interesting that it is 1.82, how many tabs do you have open, and do you use alot of PWA's or things that might use alot of local data? And I'm not going to close it yet, but theres not much I can do to help besides provide ideas, i.e. I dont know how to fix this in the thorium source.

**Comment by TrainzLuvr** [comment]
AFAIK this is just size on the disk, whether Chromium is opened or not. I think it's clear that it does not work well with network profiles, for whatever reason.

If I filed the bug with Chromium developers, I'm sure they'd just label it Wont Fix and tell me it "works as intended."

We could spend weeks and months troubleshooting this until we find a reason, but why? There's plenty of other things to do in life than searching for some edge case bug. :)

**Comment by Alex313031** [comment]
@TrainzLuvr Closing this issue. Feel free to open a new issue if you find anything interesting, find a fix for this, or wanna continue this or ask another question/file another bug.
